src/ngraph/runtime/gpu/gpu_emitter.cpp

```diff
@@ -732,8 +732,48 @@ void runtime::gpu::GPU_Emitter::emit_Max(EMIT_ARGS)
     }
 
     const ngraph::op::Max* max = static_cast<const ngraph::op::Max*>(node);
-    auto& cudnn_emitter = external_function->get_primitive_emitter()->get_cudnn_emitter();
-    auto index = cudnn_emitter->build_primitive(max);
+
+    size_t index;
+    if (args[0].get_element_type() == element::i32)
+    {
+        // one of args0 axes has zero size, zero output, use args1 value
+        if (args[0].get_size() == 0)
+        {
+            writer << out[0].get_type()
+                   << " init_value = " << TypeInfo::Get(args[0].get_type())->min() << ";\n";
+            writer << "vector<" << out[0].get_type() << "> temp(" << out[0].get_size()
+                   << ", init_value);\n";
+            writer << "runtime::gpu::cuda_memcpyHtD(" << out[0].get_name()
+                   << ", (void*)temp.data(), " << out[0].get_size() << " * "
+                   << out[0].get_element_type().size() << ");\n";
+            return;
+        }
+        else if (args[0].get_size() == out[0].get_size())
+        {
+            kernel::emit_memcpyDtD(writer, out[0], args[0]);
+            return;
+        }
+        else
+        {
+            auto axes_set = max->get_reduction_axes();
+            ngraph::AxisVector axes_vec;
+            for (auto a : axes_set)
+            {
+                axes_vec.push_back(a);
+            }
+            vector<string> dtypes;
```

Comment:
nit: can we add a small helper that takes args/out and returns a vector<string> and another one that returns vector<element_type> for each tensor in args and out? 

doesn't have to be in this PR, just thinking about it since we do this a lot

Reply:
Sure I can do that

Reply:
@csullivan I will address this in the next PR.